app/controllers/external_works_controller.rb

```diff
@@ -10,13 +10,13 @@ def new
 
   # Used with bookmark form to get an existing external work and return it via ajax
   def fetch
-   if params[:external_work_url]
-     url = Addressable::URI.heuristic_parse(params[:external_work_url]).to_str
-     @external_work = ExternalWork.where(url: url).first
-   end
-   respond_to do |format|
-    format.json { render 'fetch.js.erb' }
-   end
+    if params[:external_work_url]
+      url = Addressable::URI.heuristic_parse(params[:external_work_url]).to_str
+      @external_work = ExternalWork.where(url: url).first
+    end
+    respond_to do |format|
+      format.js
```

Comment:
Something about the way Rails resolves templates has changed in version 7, and this was easier than fighting it. The client seems not to care, and the file is JS anyways...